File: factors.c
#define _GNU_SOURCE
#include <stdio.h>
#include <stdlib.h>

void *factorize(unsigned long long int n)
{
	unsigned long i;
	/*static unsigned long r[2];*/
	/*(void)r;*/
	for (i = 2; i < n; i++)
	{
		if (n % i == 0)
		{
			printf("%llu=%lu*%lu\n", n, (unsigned long)n / i, (unsigned long)i);
			return (0);
		}
	}

	printf("%llu=%lu*%lu\n", n, (unsigned long)n, (unsigned long)1);
	return (0);
}


int main(int argc, char *argv[])
{
	FILE *fp;
	char *line = NULL, *ptr;
	size_t len = 0;
	ssize_t read;
	unsigned long *factors;
	unsigned long long int number;

	(void) factors;
	(void) ptr;

	if (argc != 2)
	{
		printf("Usage: %s <file>", argv[0]);
		exit(1);
	}

	fp = fopen(argv[1], "r");
	if (fp == NULL)
		exit(EXIT_FAILURE);

	while ((read = getline(&line, &len, fp)) != -1)
	{
		factorize(strtoull(line, &ptr, 10));
	}

	fclose(fp);
	if (line)
		free(line);
	exit(EXIT_SUCCESS);
}

File: factors.py
#!/usr/bin/python3
import sys

def factorize(num):
    """ Generate 2 factors for a given number"""
    factor1 = 2
    while (num % factor1):
        if (factor1 <= num):
            factor1 += 1
    factor2 = num // factor1
    return [factor2, factor1]

# def factorize(number):
# 	for i in range(2, number):
# 		if number % i == 0:
# 			return [i, int(number / i)]
# 	return [1, int(number)]


if __name__ == "__main__":
	if len(sys.argv) == 2:
		filename = sys.argv[1]
		with open(filename, 'r') as file:
			for line in file.readlines():
				num = int(line.strip())
				factors = factorize(num)
				print("{:d}={:d}*{:d}".format(num, factors[0], factors[1]))
	else:
		sys.exit("Usage: {} <file>".format(sys.argv[0]))
